File: androld/app/src/main/java/z3d/units/LoadManager.java
package z3d.units;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class LoadManager {

    public static String  BYTE_TYPE="BYTE_TYPE";
    public static String  IMG_TYPE="IMG_TYPE";
    public static String  XML_TYPE="XML_TYPE";


    private static LoadManager _instance;

    public static LoadManager getInstance()  {
        if (LoadManager._instance==null) {
            LoadManager._instance = new LoadManager();
        }
        return LoadManager._instance;
    }
    public List<LoaderThread> loadThreadList;
    public List<LoadInfo> waitLoadList;
    public LoadManager()
    {
        this.loadThreadList=new ArrayList<>();
        this.waitLoadList=new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            this.loadThreadList.add(new LoaderThread());
        }
    }

    public void loadUrl(String url,String typestr,  LoadBackFun backFun, Object info)
    {
        LoadInfo loadInfo = new LoadInfo();
        loadInfo.url=url;
        loadInfo.type=typestr;
        loadInfo.fun=backFun;
        loadInfo.info=info;
        for (int i = 0; i < this.loadThreadList.size(); i++) {
            if (this.loadThreadList.get(i).idle) {
                this.loadThreadList.get(i).load(loadInfo);
                return;
            }
        }

        this.waitLoadList.add(loadInfo);


    }
    public void loadWaitList()
    {
        if (this.waitLoadList.size() <= 0) {
            return;
        }
        for (int i = 0; i < this.loadThreadList.size(); i++) {
            if (this.loadThreadList.get(i).idle) {
                this.loadThreadList.get(i).load(  this.waitLoadList.remove(this.waitLoadList.size()-1));
                return;
            }
        }

    }
}

File: h5gl/new/src/pan3d/res/LoadManager.ts
module Pan3d {
    export class LoadManager {
        public static BYTE_TYPE: string = "BYTE_TYPE";
        public static IMG_TYPE: string = "IMG_TYPE";
        public static XML_TYPE: string = "XML_TYPE";

        private static _instance: LoadManager;
        public static getInstance(): LoadManager {
            if (!this._instance) {
                this._instance = new LoadManager();
            }
            return this._instance;
        }

        static getVersion(vkey): string {
            return ""
        }

        private _loadThreadList: Array<LoaderThread>;
        private _waitLoadList: Array<LoadInfo>;
        constructor() {
            this._loadThreadList = new Array;
            this._waitLoadList = new Array;

            for (var i: number = 0; i < 10; i++) {
                this._loadThreadList.push(new LoaderThread());
            }
        }

        public load($url: string, $type: string, $fun: Function, $info: any = null, $progressFun: Function = null): void {
      
            if (!$url || $url.length < 1 || $url.search("undefined") != -1) {
                //console.log("加载地址不能为空")
                return;
            }


            var version: string ="0"
     

            //GameInstance.mapName


            var loadInfo: LoadInfo = new LoadInfo($url, $type, $fun, $info, $progressFun);
            loadInfo.version = version;

            for (var i: number = 0; i < this._loadThreadList.length; i++) {
                if (this._loadThreadList[i].idle) {
                    this._loadThreadList[i].load(loadInfo);
                    return;
                }
            }

            this._waitLoadList.push(loadInfo);

        }

        public loadWaitList() {
            if (this._waitLoadList.length <= 0) {
                return;
            }

            for (var i: number = 0; i < this._loadThreadList.length; i++) {
                if (this._loadThreadList[i].idle) {
                    this._loadThreadList[i].load(this._waitLoadList.shift());
                    return;
                }
            }

        }



    }
 

    
}



